src/main/java/com/devcourse/java/calculator/InputValidator.java

```diff
@@ -0,0 +1,88 @@
+package com.devcourse.java.calculator;
+
+import com.devcourse.java.calculator.constant.ExceptionConstant;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.InputMismatchException;
+import java.util.LinkedHashMap;
+
+public class InputValidator {
+
+    public void checkCommandInput(String input) {
+        if (!isInteger(input)) {
+            throw new InputMismatchException(ExceptionConstant.COMMAND_INPUT_NOT_INTEGER_EXCEPTION);
+        }
+
+        if (!isInBoundary(input)) {
+            throw new InputMismatchException(ExceptionConstant.COMMAND_INPUT_NOT_IN_BOUNDARY_EXCEPTION);
+        }
+    }
+
+    private boolean isInteger(String input) {
+        return input.matches("^?\\d*$");
+    }
+
+    private boolean isInBoundary(String input) {
+        int command = Integer.parseInt(input);
+        return (command >= 1 && command <= 3);
+    }
+
+    public boolean isOperator(String token) {
+        return token.matches("^[\\+\\-\\*\\/]$");
+    }
+
+    public void checkCalculateHistoryLength(LinkedHashMap<Integer, String> calculateHistory) {
+        if (calculateHistory.isEmpty()) {
+            throw new IllegalArgumentException(ExceptionConstant.EMPTY_CALCULATE_HISTORY_EXCEPTION);
+        }
+    }
+
+    public void checkEquationInput(String equation) {
+        ArrayList<String> tokens = new ArrayList<>(Arrays.asList(equation.split(" ")));
+
+        throwIfEquationIsEmpty(equation);
+        throwIfEquationStartOrEndWithNoInteger(tokens);
+        throwIfEquationIsWrong(tokens);
+    }
+
+    public void throwIfEquationIsEmpty(String equation) {
+        if (equation.length() == 0) {
```

Comment:
String 내부에도 유틸 메소드가 있답니다~ 가독성이 훨씬 좋죠?

```suggestion
        if (equation.isEmpty()) {
```

Reply:
isEmpty() 잘 기억해두겠습니다!